### PR TITLE
Speedup es6 sets

### PR DESCRIPTION
**Pull request by fazouane-marouane**
Hello,

This is code for v1.0 that uses ES6 sets for the underlying lru queue. This may be merged/published after merge of #3 or #4 .

The idea behind the implementation is: "Sets are ordered by insertion order". So updating the queue is similar to:
1/ a remove/add to the Set in case of a hit
2/ remove the first element of the Set in case of an eviction due to a cache miss

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "lru-queue",
-  "version": "0.1.0",
```
Comment (medikoo) [inline package.json]: Let's not add versions (I use [standard-version](https://github.com/conventional-changelog/standard-version#readme) for that

File: package.json [hunk 2]
```diff
@@ -25,8 +25,9 @@
     "root": true
   },
   "scripts": {
+    "fix-lint": "eslint --ignore-path=.gitignore . --fix",
```
Comment (medikoo) [inline package.json]: Let's not add this script

Comment (fazouane-marouane) [inline package.json]:
```suggestion
```

File: package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "lru-queue",
-  "version": "0.1.0",
+  "version": "1.0.0",
```
Comment (fazouane-marouane) [inline package.json]:
```suggestion
  "version": "0.1.0",
```

File: package.json (hunk 1 above)
Comment (fazouane-marouane) [reply package.json]: fixed

File: package.json (hunk 2 above)
Comment (fazouane-marouane) [reply package.json]: fixed

File: package.json [hunk 4]
```diff
@@ -26,7 +26,7 @@
   },
   "scripts": {
     "lint": "eslint --ignore-path=.gitignore .",
-    "test": "node node_modules/tad/bin/tad"
+    "test": "tad index.js"
```
Comment (fazouane-marouane) [inline package.json]:
```suggestion
    "test": "node node_modules/tad/bin/tad"
```

**Pull request closed without merging**